ime/app/src/main/java/com/anysoftkeyboard/AnySoftKeyboard.java

```diff
@@ -1348,10 +1348,17 @@ public void onUpdateSelection(
     }
 
     private void updateShiftStateNow() {
+        boolean keyboardAutoCap;
```

Comment:
Instead of query for this every time this method is called, you can have it as a class member and set it inside `onAlphabetKeyboardSet()`

Reply:
Alright I moved it around!